File: packages/sdk-ts/src/clients/graph.ts
import type { HttpClient } from '../http.js';

export class GraphClient {
  constructor(private readonly http: HttpClient) {}

  /** List entities (paginated). */
  async listEntities(options?: { limit?: number; cursor?: string }): Promise<any> {
    const params = new URLSearchParams();
    if (options?.limit) params.set('limit', String(options.limit));
    if (options?.cursor) params.set('cursor', options.cursor);
    return this.http.request('GET', `/v1/entities?${params}`);
  }

  /** Get a single entity by ID. */
  async getEntity(id: string): Promise<any> {
    return this.http.request('GET', `/v1/entities/${id}`);
  }

  /** Get related entities for a given entity. */
  async getRelated(entityId: string, depth?: number): Promise<any> {
    const params = depth ? `?depth=${depth}` : '';
    return this.http.request('GET', `/v1/entities/${entityId}/graph${params}`);
  }
}

File: packages/sdk-python/steno_ai/graph.py
"""Graph client — explore entity relationships."""

from typing import Optional

from .client import HttpClient


class GraphClient:
    def __init__(self, http: HttpClient):
        self._http = http

    def list_entities(self, limit: int = 20, cursor: Optional[str] = None) -> dict:
        """List known entities."""
        params = f"limit={limit}"
        if cursor:
            params += f"&cursor={cursor}"
        return self._http.request("GET", f"/v1/entities?{params}")

    def get_entity(self, entity_id: str) -> dict:
        """Get a single entity by ID."""
        return self._http.request("GET", f"/v1/entities/{entity_id}")

    def get_related(self, entity_id: str, depth: int = 3) -> dict:
        """Get related entities up to a given depth."""
        return self._http.request("GET", f"/v1/entities/{entity_id}/graph?depth={depth}")
